server/services/datacommons.py

```diff
@@ -230,6 +230,13 @@ def get_variable_group_info(dcid: str, entities: List[str]) -> Dict:
     url += "?constrained_entities=" + "&constrained_entities=".join(entities)
   return get(url).get("info", {})
 
+def get_variable_group_info_bulk(dcids: List[str], entities: List[str]) -> Dict:
```

Comment:
instead of adding a new function, can just update the existing get_variable_group_info() to take a list, then need to change existing call site (only 1 or 2 places).

We have this pattern to use bulk version in datacommons.py and let the caller to pass a list.

Reply:
Let's discuss this offline. I believe the two APIs are returning different things:

1. non-bulk API query/response:
**https://api.datacommons.org/v1/info/variable-group/dc/g/Root?constrained_entities=geoId/06**
{'node': 'dc/g/Root',
 'info': {'absoluteName': 'Data Commons Variables',
  'childStatVarGroups': [{'id': 'dc/g/Agriculture',
    'specializedEntity': 'Agriculture',
    'displayName': 'Agriculture',
    'descendentStatVarCount': 156},
   {'id': 'dc/g/Demographics',
    'specializedEntity': 'Demographics',
    'displayName': 'Demographics',
    'descendentStatVarCount': 18948},
   {'id': 'dc/g/Economy',
    'specializedEntity': 'Economy',
    'displayName': 'Economy',
    'descendentStatVarCount': 41738},
   {'id': 'dc/g/Education',
    'specializedEntity': 'Education',
    'displayName': 'Education',
    'descendentStatVarCount': 984},
   {'id': 'dc/g/Energy',
    'specializedEntity': 'Energy',
    'displayName': 'Energy',
    'descendentStatVarCount': 2577},
   {'id': 'dc/g/Environment',
    'specializedEntity': 'Environment',
    'displayName': 'Environment',
    'descendentStatVarCount': 575},
   {'id': 'dc/g/Health',
    'specializedEntity': 'Health',
    'displayName': 'Health',
    'descendentStatVarCount': 3444},
   {'id': 'dc/g/Housing',
    'specializedEntity': 'Housing',
    'displayName': 'Housing',
    'descendentStatVarCount': 473},
   {'id': 'dc/g/Crime',
    'specializedEntity': 'Crime',
    'displayName': 'Crime',
    'descendentStatVarCount': 3421},
   {'id': 'dc/g/Uncategorized',
    'specializedEntity': 'Uncategorized',
    'displayName': 'Uncategorized',
    'descendentStatVarCount': 349}],
  'descendentStatVarCount': 52121}}

2. Bulk query/response:
**https://api.datacommons.org/v1/bulk/info/variable-group?nodes=dc/g/Root&constrained_entities=geoId/06**
{'data': [{'node': 'dc/g/Root',
   'info': {'childStatVarGroups': [
     {'id': 'dc/g/Agriculture',  'specializedEntity': 'Agriculture'},
     {'id': 'dc/g/Demographics', 'specializedEntity': 'Demographics'},
     {'id': 'dc/g/Economy', 'specializedEntity': 'Economy'},
     {'id': 'dc/g/Education', 'specializedEntity': 'Education'},
     {'id': 'dc/g/Energy', 'specializedEntity': 'Energy'},
     {'id': 'dc/g/Environment', 'specializedEntity': 'Environment'},
     {'id': 'dc/g/Health', 'specializedEntity': 'Health'},
     {'id': 'dc/g/Housing', 'specializedEntity': 'Housing'},
     {'id': 'dc/g/Crime', 'specializedEntity': 'Crime'},
     {'id': 'dc/g/Uncategorized', 'specializedEntity': 'Uncategorized'}]}}]}

There is a lot more info coming back from the non-bulk response and so I don't think we can take the bulk response and return the non-bulk response for the existing queries. The bulk response is missing information.

Reply:
Yes, I checked and the bulk API needs update. Will send you PR for that.